modin/backends/base/query_compiler.py

```diff
@@ -1569,49 +1585,77 @@ def has_multiindex(self, axis=0):
         assert axis == 1
         return isinstance(self.columns, pandas.MultiIndex)
 
-    def get_index_name(self):
+    def get_index_name(self, axis=0):
         """
-        Get index name.
+        Get index name of specified axis.
+
+        Parameters
+        ----------
+        axis: int (default 0),
+            Axis to return index name on.
 
         Returns
         -------
         hashable
             Index name, None for MultiIndex.
         """
-        return self.index.name
+        return self.get_axis(axis).name
 
-    def set_index_name(self, name):
+    def set_index_name(
+        self, name=no_default, axis=0, index=no_default, columns=no_default
```

Comment:
Please answer on follow questions:
* Why are you using `no_default` value from pandas?
* What does this mean?

Reply:
`no_default` is an object from pandas library that indicates, that this parameter was not specified, we can't use `None` here, because we may want to set the index name as `None`